integration/fuse/src/main/java/alluxio/fuse/cli/FuseShell.java

```diff
@@ -71,53 +94,36 @@ public URIStatus runCommand(AlluxioURI uri) throws InvalidArgumentException {
       throw new InvalidArgumentException("Command is needed in Fuse shell");
     }
 
-    String cmdType = cmds[2];
-    switch (CommandType.fromValue(cmdType)) {
-      case METADATA_CACHE:
-        MetadataCacheCommand command = new MetadataCacheCommand(mFileSystem, mConf);
-        return command.run(path, Arrays.copyOfRange(cmds, 3, cmds.length));
-      default:
-        throw new InvalidArgumentException("Fuse shell command not found");
+    FuseCommand command = mCommands.get(cmds[2]);
+    if (command == null) {
+      logUsage();
+      throw new InvalidArgumentException(String.format("%s is an unknown command.", cmds[2]));
     }
+    String [] currArgs = Arrays.copyOfRange(cmds, 2, cmds.length);
+    try {
+      while (command.hasSubCommand()) {
+        if (currArgs.length < 2) {
+          throw new InvalidArgumentException("No sub-command is specified");
+        }
+        if (!command.getSubCommands().containsKey(currArgs[1])) {
```

Comment:
Did you test the new Fuse shell out?

Reply:
Just checking

Reply:
Test on my local environment and all manual test is passed.